Sources/NIOSSL/UniversalBootstrapSupport.swift

```diff
@@ -0,0 +1,44 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+public struct NIOSSLClientTLSProvider<Bootstrap: NIOClientTCPBootstrapProtocol>: NIOClientTLSProvider {
+    public typealias Bootstrap = Bootstrap
+
+    let context: NIOSSLContext
+    let serverHostname: String?
+    let customVerificationCallback: NIOSSLCustomVerificationCallback?
+
+    public init(context: NIOSSLContext,
+                serverHostname: String?,
+                customVerificationCallback: NIOSSLCustomVerificationCallback? = nil) {
+        self.context = context
+        self.serverHostname = serverHostname
+        self.customVerificationCallback = customVerificationCallback
+    }
+
+    public func enableTLS(_ bootstrap: Bootstrap) -> Bootstrap {
+        return bootstrap.protocolHandlers({
```

Comment:
Any reason not not use trailing closure syntax here?

Reply:
no, will fix